.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,48 @@
+# *********************************** WARNING **************************************
+# We have not automatically identified the languages in your repository.
+# Please edit the `language:` list below to add the appropriate supported languages.
+# **********************************************************************************
+name: "CodeQL"
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [main]
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ['go']
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        # We must fetch at least the immediate parents so that if this is
+        # a pull request then we can checkout the head.
+        fetch-depth: 2
+
+    # If this run was triggered by a pull request event, then checkout
+    # the head of the pull request instead of the merge commit.
+    - run: git checkout HEAD^2
```

Comment:
If it works, I'm fine, but why is the "actual" head of a PR branch 2 commits behind? I thought it was just behind the merge commit?